holoviews/core/data/xarray.py

```diff
@@ -167,24 +178,49 @@ def groupby(cls, dataset, dimensions, container_type, group_type, **kwargs):
 
 
     @classmethod
-    def coords(cls, dataset, dim, ordered=False, expanded=False):
-        if expanded:
-            return util.expand_grid_coords(dataset, dim)
+    def coords(cls, dataset, dimension, ordered=False, expanded=False, edges=False):
+        dim = dataset.get_dimension(dimension)
+        dim = dimension if dim is None else dim.name
+        irregular = cls.irregular(dataset, dim)
+        if irregular or expanded:
+            if irregular:
+                data = dataset.data[dim]
+            else:
+                data = util.expand_grid_coords(dataset, dim)
+            if edges:
+                data = cls._infer_interval_breaks(data, axis=1)
+                data = cls._infer_interval_breaks(data, axis=0)
+            return data
+
         data = np.atleast_1d(dataset.data[dim].data)
         if ordered and data.shape and np.all(data[1:] < data[:-1]):
             data = data[::-1]
+        shape = cls.shape(dataset, True)
+
+        if dim in dataset.kdims:
+            idx = dataset.get_dimension_index(dim)
+            isedges = (dim in dataset.kdims and len(shape) == dataset.ndims
+                       and len(data) == (shape[dataset.ndims-idx-1]+1))
+        else:
+            isedges = False
+        if edges and not isedges:
+            data = cls._infer_interval_breaks(data)
+        elif not edges and isedges:
+            data = np.convolve(data, [0.5, 0.5], 'valid')
```

Comment:
Why is there a convolution here?

Reply:
Easiest way to compute a rolling mean, i.e. edges -> bin centers.